src/App.tsx

```diff
@@ -1,61 +1,121 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [userError, setUserError] = useState('');
+  const [visibleTodos, setVisibleTodos] = useState([...todos]);
+
+  const generateTodoId = (currentTodos: Todo[]) => {
+    const biggestId = [...currentTodos].sort((previous, current) => (
+      current.id - previous.id
+    ))[0].id;
+
+    return biggestId + 1;
+  };
```

Comment:
It isn't a good method for generating an id. You can use different libraries for it